tensorly/backend/paddle_backend.py

```diff
@@ -0,0 +1,366 @@
+        if axis is None:
+            axis = tuple(range(tensor.ndim))
+        return paddle.sum(tensor, axis=axis, keepdim=keepdims)
+
+    @staticmethod
+    def max(tensor: paddle.Tensor, axis=None):
+        if axis is None:
+            return paddle.max(tensor)
+        else:
+            return paddle.max(tensor, axis=axis)[0]
+
+    @staticmethod
+    def flip(tensor: paddle.Tensor, axis=None):
+        if isinstance(axis, int):
+            axis = [axis]
+
+        if axis is None:
+            return paddle.flip(tensor, axis=[i for i in range(tensor.ndim)])
+        else:
+            return paddle.flip(tensor, axis=axis)
+
+    @staticmethod
+    def concatenate(tensors: paddle.Tensor, axis=0):
+        return paddle.concat(tensors, axis=axis)
+
+    @staticmethod
+    def argmin(input: paddle.Tensor, axis=None):
+        return paddle.argmin(input, axis=axis)
+
+    @staticmethod
+    def argsort(input: paddle.Tensor, axis=None):
+        return paddle.argsort(input, axis=axis)
+
+    @staticmethod
+    def argmax(input: paddle.Tensor, axis=None):
+        return paddle.argmax(input, axis=axis)
+
+    @staticmethod
+    def stack(arrays: paddle.Tensor, axis=0):
+        return paddle.stack(arrays, axis=axis)
+
+    @staticmethod
+    def diag(tensor: paddle.Tensor, k=0):
+        return paddle.diag(tensor, offset=k)
+
+    @staticmethod
+    def sort(tensor: paddle.Tensor, axis):
+        if axis is None:
+            tensor = tensor.flatten()
+            axis = -1
+
+        return paddle.sort(tensor, axis=axis)
+
+    @staticmethod
+    def update_index(tensor: paddle.Tensor, index, values):
+        tensor.index_put_(index, values)
+
+    @staticmethod
+    def lstsq(a: paddle.Tensor, b: paddle.Tensor, rcond=None, driver="gelsd"):
+        # NOTE: Behavior in paddle.linalg.lstsq may be different from numpy.linalg.lstsq
+        # so we add some extra process here to keep the same behavior.
+        vector_case = b.ndim == 1 or b.ndim == a.ndim - 1
+        if vector_case:
+            b = b.unsqueeze(-1)
+
+        m, n = a.shape[-2], a.shape[-1]
+        sol, res, rank, single_value = paddle.linalg.lstsq(
+            a, b, rcond=rcond, driver=driver
+        )
+        if m > n and driver != "gelsy":
+            compute_residuals = True
+            if driver in ["gelss", "gelsd"]:
+                if a.ndim == 2:
+                    compute_residuals = rank.item() == n
+                else:
+                    compute_residuals = all([r.item() == n for r in rank])
+            if not compute_residuals:
+                res = paddle.empty([0], dtype=a.dtype)
+
+        if vector_case:
+            sol = sol.squeeze(-1)
+        return sol, res, rank, single_value
+
+    @staticmethod
+    def eigh(tensor: paddle.Tensor):
+        return paddle.linalg.eigh(tensor)
+
+    @staticmethod
+    def sign(tensor: paddle.Tensor):
+        """paddle.sign does not support complex numbers."""
```

Comment:
Indeed, I'm not seeing [`paddle.sgn`](https://www.paddlepaddle.org.cn/en/searchall?q=paddle.sgn&language=en&version=2.2) in the documentation.

Reply:
1. The english document will be fixed as soon as possible, https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/sgn_en.html
2. `paddle.sgn has` same behavior and degisning motivation as pytorch, which returns `x/abs(x)` rather than `-1/1`